lib/tbot/workloadidentity/workloadattest/podman/client.go

```diff
@@ -0,0 +1,164 @@
+/*
+ * Teleport
+ * Copyright (C) 2025  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+// Package podman implements bindings for a small subset of Podman's REST API.
+//
+// We use this instead of importing github.com/containers/podman/v5/pkg/bindings
+// because we only use a couple of endpoints, so it's not worth taking on its
+// transitive dependencies.
+package podman
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net"
+	"net/http"
+	"net/url"
+
+	"github.com/gravitational/trace"
+)
+
+// APIVersion is the version of the Podman REST API we support.
+//
+// Note: Podman doesn't really seem to check or enforce the version segment
+// of the URL path, so setting it doesn't have much of an effect.
+//
+// https://github.com/containers/podman/blob/62cde17193e1469de15995ba78bd909fd07dffc6/pkg/api/server/handler_api.go#L69-L74
+const APIVersion = "v4.0.0"
+
+// Client for Podman's HTTP REST API.
+type Client struct{ httpClient *http.Client }
+
+// NewClient creates a Client for the API service at the given address.
+//
+// Note: addr must be in the form `unix://path/to/socket`, we do not currently
+// support connecting to the Podman service over TCP because it's generally
+// insecure.
+func NewClient(addr string) (*Client, error) {
+	u, err := url.Parse(addr)
+	if err != nil {
+		return nil, trace.Wrap(err, "invalid address: %q", addr)
+	}
+	if u.Scheme != "unix" {
+		return nil, trace.BadParameter("only unix domain sockets are supported")
+	}
+	return &Client{
+		httpClient: &http.Client{
+			Transport: &http.Transport{
+				DialContext: func(ctx context.Context, _ string, _ string) (net.Conn, error) {
+					return (&net.Dialer{}).DialContext(ctx, "unix", u.Path)
+				},
+			},
+		},
+	}, nil
+}
+
+// InspectContainer reads information about the container with the given ID.
+//
+// https://docs.podman.io/en/latest/_static/api.html#tag/containers/operation/ContainerInspectLibpod
+func (c *Client) InspectContainer(ctx context.Context, id string) (*Container, error) {
+	rsp, err := c.get(ctx, fmt.Sprintf("containers/%s", id))
+	if err != nil {
+		return nil, err
+	}
+	defer rsp.Body.Close()
+
+	var cntr Container
+	if err := json.NewDecoder(rsp.Body).Decode(&cntr); err != nil {
+		return nil, trace.Wrap(err, "failed to decode container response")
```

Comment:
Nit: I avoid using words like "failed" when wrapping an error since after a few layers, you end up with "failed" several times in the message. I'd prefer something like "decoding container response".